addons/product/models/product.py

```diff
@@ -620,9 +620,14 @@ def _select_seller(self, partner_id=False, quantity=0.0, date=None, uom_id=False
             if seller.product_id and seller.product_id != self:
                 continue
 
-            res |= seller
-            break
-        return res
+            res += seller
```

Comment:
@hch-odoo remove your diff and .sorted('price') on sellers

Reply:
as per the document https://docs.google.com/document/d/10StevS6AIPnCHUKRmT11GumsifgD9z-FTd6jubpPAj0/edit
we have to consider the sequence of the suppliers first and then we need to check for the ordered quantity.
as you can check in the document, if there are multiple suppliers for the same product but having different price based on ordered quantity, then we have to consider the quantity.

Reply:
@hch-odoo then it's an order by quantity desc and sequence but this specification seems strange and hard to understand for classic user, I will check with Laura

Reply:
add a condition in this case
``` python
if not seller or seller.name == res.name:
    res |= seller
# after the loop
return res.sorted('price')[0]
```